src/Lecture2.hs

```diff
@@ -70,7 +73,15 @@ return the removed element.
 >>> removeAt 10 [1 .. 5]
 (Nothing,[1,2,3,4,5])
 -}
-removeAt = error "TODO"
+removeAt :: (Eq a) => Int -> [a] -> (Maybe a, [a])
+removeAt index list
+  | index < 0 = (Nothing, list)
+  | list == [] = (Nothing, list)
```

Comment:
It's better to use the `null` function to check lists on emptiness. Because you're using `== []`, Haskell forces you to add the `Eq a` constraint. This means that your implementation of `removeAt` doesn't work with lists of functions for example.

However, this `Eq` constraint is not needed. You can use `null` instead and it doesn't require `Eq`.

Even better, you can pattern match on list to avoid calling `null` and avoid unsafe functions `head` and `tail`